2025-02-21 09:28:26.902002: I external/local_tsl/tsl/cuda/cudart_stub.cc:31] Could not find cuda drivers on your machine, GPU will not be used.
2025-02-21 09:28:26.936382: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2025-02-21 09:28:26.936423: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2025-02-21 09:28:26.937461: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2025-02-21 09:28:26.942852: I external/local_tsl/tsl/cuda/cudart_stub.cc:31] Could not find cuda drivers on your machine, GPU will not be used.
2025-02-21 09:28:26.943005: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2025-02-21 09:28:27.465474: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
online arguments=['play.py', '-e', 'DaisoSokcho', '-a', 'CQL_SAC_wo_normalize', '-t', '5000', '-l', '10000', '-c', '/home/soh/work/try_tf_agents/result/DaisoSokcho_SAC_0221_051416/model']
2025-02-21 09:28:28,858 - INFO - config={'projectPath': '/home/soh/work/try_tf_agents', 'dtype': 'float32', 'intdtype': 'int32', 'resultDir': 'result', 'isGpuUsed': False, 'log_level_name': 'INFO', '# log_level_name': 'like INFO or DEBUG', 'isSummaryWriterUsed': False, 'environment': 'CartPole-v0', 'environment_wrapper': None, 'agent': 'DQN', 'replaybuffer': 'tf_uniform', 'isPerUsed': False, 'driver': 'dynamic_step', 'num_actions': 3, '# num_actions discretized: for ActionDiscretizeWrapper': None, 'num_time_steps': 10000, '# num_time_steps': '0 for sys.maxsize', 'num_time_steps_to_collect': 1, 'num_time_steps_to_train': 1, 'num_time_steps_to_log': 10, 'num_time_steps_to_eval': 100, 'num_train_steps_to_save': 10000, '# num_logs': 500, '# num_evals': 50, '# for CDQN': None, 'num_atoms': 51, 'min_q_value': -20, 'max_q_value': 20, 'n_step_update': 1, 'qnet_fc_layer_params': [128, 64], 'actor_fc_layer_params': [256, 256], 'critic_observation_fc_layer_params': None, '# critic_observation_fc_layer_params': '[32, 32] for Pendulum', 'critic_action_fc_layer_params': None, '# critic_action_fc_layer_params': '[32, 32] for Pendulum', 'critic_joint_fc_layer_params': [256, 256], '# actor_fc_layer_params': '[32, 32] for Pendulum', '# critic_joint_fc_layer_params': '[128, 16] for Pendulum', 'batch_size': 64, 'learning_rate': 0.001, '# learning_rate default': 0.001, 'actor_learning_rate': 0.0003, 'critic_learning_rate': 0.0006, 'alpha_learning_rate': 0.0003, 'target_update_tau': 0.005, 'target_update_period': 1, 'gamma': 0.99, 'reward_scale_factor': 1.0, 'epsilon_greedy': 0.1, 'replaybuffer_max_length': 15000, '# num_frames = capacity = max_length * env.batch_size and default env.batch_size = 1': None, 'num_env_steps_to_collect_init': 5000, 'num_env_steps_to_collect_per_time_step': 1, 'num_episodes_to_collect_init': 10, 'num_episodes_to_collect_per_time_step': 1, 'checkpoint_max_to_keep': 2, '# for PPO': None, 'num_episodes_to_eval': 10, 'value_fc_layer_params': [256, 256], 'num_parallel_envs': 30, 'num_env_steps': 20000, 'num_collect_episodes_per_train_step': 30, '# also for BC, CQL_SAC': None, 'num_epochs': 25, 'reverb_port': 8008, '#  ending': None}
2025-02-21 09:28:28,858 - INFO - args=Namespace(environment='DaisoSokcho', environment_wrapper=None, agent='CQL_SAC_wo_normalize', replaybuffer=None, driver=None, checkpoint_path='/home/soh/work/try_tf_agents/result/DaisoSokcho_SAC_0221_051416/model', fill_after_restore=None, reverb_checkpoint_path=None, num_actions=None, num_time_steps=5000, replaybuffer_max_length=10000, num_env_steps_to_collect_init=None, epsilon_greedy=None, reverb_port=None)
2025-02-21 09:28:28,858 - INFO - environment=DaisoSokcho
2025-02-21 09:28:28,858 - INFO - envWrapper=None
2025-02-21 09:28:28,858 - INFO - agent=CQL_SAC_wo_normalize
2025-02-21 09:28:28,858 - INFO - replaybuffer=tf_uniform
2025-02-21 09:28:28,858 - INFO - driver=dynamic_step
2025-02-21 09:28:28,858 - INFO - num_env_steps_to_collect_init=5000
2025-02-21 09:28:28,858 - INFO - epsilon_greedy=0.1
2025-02-21 09:28:28,858 - INFO - checkpointPath_toRestore=/home/soh/work/try_tf_agents/result/DaisoSokcho_SAC_0221_051416/model
2025-02-21 09:28:28,858 - INFO - checkpointPath_toSave=/home/soh/work/try_tf_agents/result/DaisoSokcho_CQL_SAC_wo_normalize_0221_092828/model
2025-02-21 09:28:28,858 - INFO - reverb_port=8008
Unit Timestep : 10 min
               timestamp        date  days  ...  E_ehp_5   T_oa_max   T_oa_min
0    2023-03-23 08:00:00  2023-03-23    81  ...      0.0  13.145000   6.873333
1    2023-03-23 08:10:00  2023-03-23    81  ...      0.0  13.145000   6.873333
2    2023-03-23 08:20:00  2023-03-23    81  ...      0.0  13.145000   6.873333
3    2023-03-23 08:30:00  2023-03-23    81  ...      0.0  13.145000   6.873333
4    2023-03-23 08:40:00  2023-03-23    81  ...      0.0  13.145000   6.873333
...                  ...         ...   ...  ...      ...        ...        ...
2683 2023-06-21 21:10:00  2023-06-21   171  ...  15339.0  18.151667  18.018333
2684 2023-06-21 21:20:00  2023-06-21   171  ...  27858.0  18.118333  18.018333
2685 2023-06-21 21:30:00  2023-06-21   171  ...  29052.0  18.085000  18.018333
2686 2023-06-21 21:40:00  2023-06-21   171  ...   4897.0  18.051667  18.018333
2687 2023-06-21 21:50:00  2023-06-21   171  ...     24.0  18.018333  18.018333

[2688 rows x 25 columns]
[Data INFO] #episodes(dates): 32 (#train: 28, #test: 4)

Unit Timestep : 10 min
               timestamp        date  days  ...  E_ehp_5   T_oa_max   T_oa_min
0    2023-03-23 08:00:00  2023-03-23    81  ...      0.0  13.145000   6.873333
1    2023-03-23 08:10:00  2023-03-23    81  ...      0.0  13.145000   6.873333
2    2023-03-23 08:20:00  2023-03-23    81  ...      0.0  13.145000   6.873333
3    2023-03-23 08:30:00  2023-03-23    81  ...      0.0  13.145000   6.873333
4    2023-03-23 08:40:00  2023-03-23    81  ...      0.0  13.145000   6.873333
...                  ...         ...   ...  ...      ...        ...        ...
2683 2023-06-21 21:10:00  2023-06-21   171  ...  15339.0  18.151667  18.018333
2684 2023-06-21 21:20:00  2023-06-21   171  ...  27858.0  18.118333  18.018333
2685 2023-06-21 21:30:00  2023-06-21   171  ...  29052.0  18.085000  18.018333
2686 2023-06-21 21:40:00  2023-06-21   171  ...   4897.0  18.051667  18.018333
2687 2023-06-21 21:50:00  2023-06-21   171  ...     24.0  18.018333  18.018333

[2688 rows x 25 columns]
[Data INFO] #episodes(dates): 32 (#train: 28, #test: 4)

2025-02-21 09:28:29,465 - INFO - tf_observation_spec: BoundedTensorSpec(shape=(13,), dtype=tf.float32, name='observation', minimum=array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.], dtype=float32), maximum=array([ 365., 1440.,   40.,   40.,   40.,   40.,   40.,   40.,   40.,   40.,   10.,   60.,  400.], dtype=float32))
2025-02-21 09:28:29,465 - INFO - tf_action_spec: BoundedTensorSpec(shape=(5,), dtype=tf.float32, name='action', minimum=array([0., 0., 0., 0., 0.], dtype=float32), maximum=array([5., 7., 5., 5., 5.], dtype=float32))
2025-02-21 09:28:29,466 - INFO - tf_env_step_spec: TimeStep(
{'step_type': TensorSpec(shape=(), dtype=tf.int32, name='step_type'),
 'reward': TensorSpec(shape=(), dtype=tf.float32, name='reward'),
 'discount': BoundedTensorSpec(shape=(), dtype=tf.float32, name='discount', minimum=array(0., dtype=float32), maximum=array(1., dtype=float32)),
 'observation': BoundedTensorSpec(shape=(13,), dtype=tf.float32, name='observation', minimum=array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.], dtype=float32), maximum=array([ 365., 1440.,   40.,   40.,   40.,   40.,   40.,   40.,   40.,   40.,   10.,   60.,  400.], dtype=float32))})
2025-02-21 09:28:29,466 - INFO - min_q_value=-600, max_q_value=0
2025-02-21 09:28:29,678 - INFO - tf_agent_collect_data_spec: Trajectory(
{'step_type': TensorSpec(shape=(), dtype=tf.int32, name='step_type'),
 'observation': BoundedTensorSpec(shape=(13,), dtype=tf.float32, name='observation', minimum=array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.], dtype=float32), maximum=array([ 365., 1440.,   40.,   40.,   40.,   40.,   40.,   40.,   40.,   40.,   10.,   60.,  400.], dtype=float32)),
 'action': BoundedTensorSpec(shape=(5,), dtype=tf.float32, name='action', minimum=array([0., 0., 0., 0., 0.], dtype=float32), maximum=array([5., 7., 5., 5., 5.], dtype=float32)),
 'policy_info': (),
 'next_step_type': TensorSpec(shape=(), dtype=tf.int32, name='step_type'),
 'reward': TensorSpec(shape=(), dtype=tf.float32, name='reward'),
 'discount': BoundedTensorSpec(shape=(), dtype=tf.float32, name='discount', minimum=array(0., dtype=float32), maximum=array(1., dtype=float32))})
WARNING:tensorflow:From /home/soh/venvtfa/lib/python3.10/site-packages/tensorflow/python/autograph/impl/api.py:377: ReplayBuffer.get_next (from tf_agents.replay_buffers.replay_buffer) is deprecated and will be removed in a future version.
Instructions for updating:
Use `as_dataset(..., single_deterministic_pass=False) instead.
2025-02-21 09:28:29,775 - WARNING - From /home/soh/venvtfa/lib/python3.10/site-packages/tensorflow/python/autograph/impl/api.py:377: ReplayBuffer.get_next (from tf_agents.replay_buffers.replay_buffer) is deprecated and will be removed in a future version.
Instructions for updating:
Use `as_dataset(..., single_deterministic_pass=False) instead.
2025-02-21 09:28:30,283 - INFO - ending get_replaybuffer(), replaybuffer.num_frames()=0
2025-02-21 09:28:30,284 - INFO - No checkpoint available at /home/soh/work/try_tf_agents/result/DaisoSokcho_CQL_SAC_wo_normalize_0221_092828/model
2025-02-21 09:28:51,663 - INFO - random_policy avg_return=-321.148
2025-02-21 09:28:51,663 - INFO - replaybuffer.capacity=10000
2025-02-21 09:28:51,664 - INFO - before restoring with checkpointer, replaybuffer.num_frames()=0
2025-02-21 09:28:51,664 - INFO - restoring from checkpointPath_toRestore=/home/soh/work/try_tf_agents/result/DaisoSokcho_SAC_0221_051416/model
2025-02-21 09:28:51,664 - INFO - restoring replaybuffer
2025-02-21 09:28:51,664 - INFO - Checkpoint available: /home/soh/work/try_tf_agents/result/DaisoSokcho_SAC_0221_051416/model/ckpt-5000
2025-02-21 09:28:51,670 - INFO - after restoring with checkpointer, replaybuffer.num_frames()=10000
2025-02-21 09:28:51,670 - INFO - restoring time = 0.007
WARNING:tensorflow:Detecting that an object or model or tf.train.Checkpoint is being deleted with unrestored values. See the following logs for the specific values in question. To silence these warnings, use `status.expect_partial()`. See https://www.tensorflow.org/api_docs/python/tf/train/Checkpoint#restorefor details about the status object returned by the restore function.
2025-02-21 09:28:51,671 - WARNING - Detecting that an object or model or tf.train.Checkpoint is being deleted with unrestored values. See the following logs for the specific values in question. To silence these warnings, use `status.expect_partial()`. See https://www.tensorflow.org/api_docs/python/tf/train/Checkpoint#restorefor details about the status object returned by the restore function.
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).global_step
2025-02-21 09:28:51,671 - WARNING - Value in checkpoint could not be found in the restored object: (root).global_step
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._log_alpha
2025-02-21 09:28:51,671 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._log_alpha
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer.iter
2025-02-21 09:28:51,671 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer.iter
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer.beta_1
2025-02-21 09:28:51,671 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer.beta_1
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer.beta_2
2025-02-21 09:28:51,671 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer.beta_2
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer.decay
2025-02-21 09:28:51,671 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer.decay
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer.learning_rate
2025-02-21 09:28:51,671 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer.learning_rate
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer.iter
2025-02-21 09:28:51,671 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer.iter
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer.beta_1
2025-02-21 09:28:51,671 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer.beta_1
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer.beta_2
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer.beta_2
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer.decay
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer.decay
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer.learning_rate
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer.learning_rate
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._alpha_optimizer.iter
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._alpha_optimizer.iter
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._alpha_optimizer.beta_1
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._alpha_optimizer.beta_1
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._alpha_optimizer.beta_2
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._alpha_optimizer.beta_2
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._alpha_optimizer.decay
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._alpha_optimizer.decay
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._alpha_optimizer.learning_rate
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._alpha_optimizer.learning_rate
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._update_target._counter
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._update_target._counter
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).policy._observation_normalizer._count.0
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).policy._observation_normalizer._count.0
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).policy._observation_normalizer._avg.0
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).policy._observation_normalizer._avg.0
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).policy._observation_normalizer._m2.0
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).policy._observation_normalizer._m2.0
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).policy._observation_normalizer._m2_carry.0
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).policy._observation_normalizer._m2_carry.0
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_network_1._joint_layers.1.kernel
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_network_1._joint_layers.1.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_network_1._joint_layers.1.bias
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_network_1._joint_layers.1.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_network_1._joint_layers.2.kernel
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_network_1._joint_layers.2.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_network_1._joint_layers.2.bias
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_network_1._joint_layers.2.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_network_1._joint_layers.3.kernel
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_network_1._joint_layers.3.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_network_1._joint_layers.3.bias
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_network_1._joint_layers.3.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_network_2._joint_layers.1.kernel
2025-02-21 09:28:51,672 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_network_2._joint_layers.1.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_network_2._joint_layers.1.bias
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_network_2._joint_layers.1.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_network_2._joint_layers.2.kernel
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_network_2._joint_layers.2.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_network_2._joint_layers.2.bias
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_network_2._joint_layers.2.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_network_2._joint_layers.3.kernel
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_network_2._joint_layers.3.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_network_2._joint_layers.3.bias
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_network_2._joint_layers.3.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_1._joint_layers.1.kernel
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_1._joint_layers.1.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_1._joint_layers.1.bias
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_1._joint_layers.1.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_1._joint_layers.2.kernel
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_1._joint_layers.2.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_1._joint_layers.2.bias
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_1._joint_layers.2.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_1._joint_layers.3.kernel
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_1._joint_layers.3.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_1._joint_layers.3.bias
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_1._joint_layers.3.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_2._joint_layers.1.kernel
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_2._joint_layers.1.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_2._joint_layers.1.bias
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_2._joint_layers.1.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_2._joint_layers.2.kernel
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_2._joint_layers.2.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_2._joint_layers.2.bias
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_2._joint_layers.2.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_2._joint_layers.3.kernel
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_2._joint_layers.3.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_2._joint_layers.3.bias
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._target_critic_network_2._joint_layers.3.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_network._projection_networks._projection_layer.kernel
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_network._projection_networks._projection_layer.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_network._projection_networks._projection_layer.bias
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_network._projection_networks._projection_layer.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._train_policy._observation_normalizer._count.0
2025-02-21 09:28:51,673 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._train_policy._observation_normalizer._count.0
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._train_policy._observation_normalizer._avg.0
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._train_policy._observation_normalizer._avg.0
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._train_policy._observation_normalizer._m2.0
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._train_policy._observation_normalizer._m2.0
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._train_policy._observation_normalizer._m2_carry.0
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._train_policy._observation_normalizer._m2_carry.0
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_network._encoder._postprocessing_layers.1.kernel
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_network._encoder._postprocessing_layers.1.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_network._encoder._postprocessing_layers.1.bias
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_network._encoder._postprocessing_layers.1.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_network._encoder._postprocessing_layers.2.kernel
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_network._encoder._postprocessing_layers.2.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_network._encoder._postprocessing_layers.2.bias
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_network._encoder._postprocessing_layers.2.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'm' for (root).agent._actor_network._projection_networks._projection_layer.kernel
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'm' for (root).agent._actor_network._projection_networks._projection_layer.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'm' for (root).agent._actor_network._projection_networks._projection_layer.bias
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'm' for (root).agent._actor_network._projection_networks._projection_layer.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'm' for (root).agent._actor_network._encoder._postprocessing_layers.1.kernel
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'm' for (root).agent._actor_network._encoder._postprocessing_layers.1.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'm' for (root).agent._actor_network._encoder._postprocessing_layers.1.bias
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'm' for (root).agent._actor_network._encoder._postprocessing_layers.1.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'm' for (root).agent._actor_network._encoder._postprocessing_layers.2.kernel
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'm' for (root).agent._actor_network._encoder._postprocessing_layers.2.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'm' for (root).agent._actor_network._encoder._postprocessing_layers.2.bias
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'm' for (root).agent._actor_network._encoder._postprocessing_layers.2.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'v' for (root).agent._actor_network._projection_networks._projection_layer.kernel
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'v' for (root).agent._actor_network._projection_networks._projection_layer.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'v' for (root).agent._actor_network._projection_networks._projection_layer.bias
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'v' for (root).agent._actor_network._projection_networks._projection_layer.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'v' for (root).agent._actor_network._encoder._postprocessing_layers.1.kernel
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'v' for (root).agent._actor_network._encoder._postprocessing_layers.1.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'v' for (root).agent._actor_network._encoder._postprocessing_layers.1.bias
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'v' for (root).agent._actor_network._encoder._postprocessing_layers.1.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'v' for (root).agent._actor_network._encoder._postprocessing_layers.2.kernel
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'v' for (root).agent._actor_network._encoder._postprocessing_layers.2.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'v' for (root).agent._actor_network._encoder._postprocessing_layers.2.bias
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._actor_optimizer's state 'v' for (root).agent._actor_network._encoder._postprocessing_layers.2.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_1._joint_layers.1.kernel
2025-02-21 09:28:51,674 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_1._joint_layers.1.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_1._joint_layers.1.bias
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_1._joint_layers.1.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_1._joint_layers.2.kernel
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_1._joint_layers.2.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_1._joint_layers.2.bias
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_1._joint_layers.2.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_1._joint_layers.3.kernel
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_1._joint_layers.3.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_1._joint_layers.3.bias
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_1._joint_layers.3.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_2._joint_layers.1.kernel
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_2._joint_layers.1.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_2._joint_layers.1.bias
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_2._joint_layers.1.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_2._joint_layers.2.kernel
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_2._joint_layers.2.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_2._joint_layers.2.bias
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_2._joint_layers.2.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_2._joint_layers.3.kernel
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_2._joint_layers.3.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_2._joint_layers.3.bias
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'm' for (root).agent._critic_network_2._joint_layers.3.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_1._joint_layers.1.kernel
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_1._joint_layers.1.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_1._joint_layers.1.bias
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_1._joint_layers.1.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_1._joint_layers.2.kernel
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_1._joint_layers.2.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_1._joint_layers.2.bias
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_1._joint_layers.2.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_1._joint_layers.3.kernel
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_1._joint_layers.3.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_1._joint_layers.3.bias
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_1._joint_layers.3.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_2._joint_layers.1.kernel
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_2._joint_layers.1.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_2._joint_layers.1.bias
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_2._joint_layers.1.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_2._joint_layers.2.kernel
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_2._joint_layers.2.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_2._joint_layers.2.bias
2025-02-21 09:28:51,675 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_2._joint_layers.2.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_2._joint_layers.3.kernel
2025-02-21 09:28:51,676 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_2._joint_layers.3.kernel
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_2._joint_layers.3.bias
2025-02-21 09:28:51,676 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._critic_optimizer's state 'v' for (root).agent._critic_network_2._joint_layers.3.bias
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._alpha_optimizer's state 'm' for (root).agent._log_alpha
2025-02-21 09:28:51,676 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._alpha_optimizer's state 'm' for (root).agent._log_alpha
WARNING:tensorflow:Value in checkpoint could not be found in the restored object: (root).agent._alpha_optimizer's state 'v' for (root).agent._log_alpha
2025-02-21 09:28:51,676 - WARNING - Value in checkpoint could not be found in the restored object: (root).agent._alpha_optimizer's state 'v' for (root).agent._log_alpha
2025-02-21 09:29:30,791 - INFO - before training, avg_return=-396.167
Traceback (most recent call last):
  File "/home/soh/work/try_tf_agents/src/play.py", line 862, in <module>
    game.run(logger, py_train_env, tf_eval_env, agent, replaybuffer, iterator, driver, checkpointer_toSave, reverb_client_toSave)
  File "/home/soh/work/try_tf_agents/src/game.py", line 100, in run
    loss_info = agent.train(trajectory)
  File "/home/soh/venvtfa/lib/python3.10/site-packages/tensorflow/python/util/traceback_utils.py", line 153, in error_handler
    raise e.with_traceback(filtered_tb) from None
  File "/home/soh/venvtfa/lib/python3.10/site-packages/tensorflow/python/eager/execute.py", line 53, in quick_execute
    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
tensorflow.python.framework.errors_impl.InvalidArgumentError: Graph execution error:

Detected at node CheckNumerics_1 defined at (most recent call last):
  File "/home/soh/work/try_tf_agents/src/play.py", line 862, in <module>

  File "/home/soh/work/try_tf_agents/src/game.py", line 100, in run

  File "/home/soh/venvtfa/lib/python3.10/site-packages/tf_agents/agents/tf_agent.py", line 348, in train

  File "/home/soh/venvtfa/lib/python3.10/site-packages/tf_agents/utils/common.py", line 193, in with_check_resource_vars

  File "/home/soh/venvtfa/lib/python3.10/site-packages/tf_agents/agents/cql/cql_sac_agent.py", line 295, in _train

CQL loss is inf or nan. : Tensor had NaN values
	 [[{{node CheckNumerics_1}}]] [Op:__inference_train_135988]
